api/lib/opentelemetry/trace/span_id.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Trace
+    # A class that represents a trace identifier. A valid span identifier is an
+    # 8-byte array with at least one non-zero byte.
+    class SpanId
+      INVALID_ID = 0
+      private_constant :INVALID_ID
+
+      MAX_64BIT_UNSIGNED_INT = (1 << 64) - 1
+      private_constant :MAX_64BIT_UNSIGNED_INT
+
+      def self.generate
+        id = rand(MAX_64BIT_UNSIGNED_INT)
```

Comment:
In Shopify’s tracing gem, we limited the span IDs to the range `1..(2**62 - 1)`, sacrificing 2 bits to ensure we only generated Fixnums (avoiding extra allocations for Bignums).

Reply:
I don't think it's a problem since ruby 2.4 when Bignum and Fixnum were merged into Integer.
```
0.size # => 8
((1 << 64) - 1).size # => 8
```